core/types/receipt.go

```diff
@@ -507,8 +545,16 @@ func (rs Receipts) EncodeIndex(i int, w *bytes.Buffer) {
 		}
 	case DepositTxType:
 		w.WriteByte(DepositTxType)
-		if err := rlp.Encode(w, data); err != nil {
-			panic(err)
+		if r.DepositReceiptVersion != nil {
```

Comment:
https://github.com/testinprod-io/op-erigon/pull/99/files/3e319d2f63d7bdae176263451208644ee7b7e559#r1384338363 
^ Even if we don't need DepositREceiptVersion nil check at the comment, do we still need this nil check here?

Reply:
This is for backward compatibility. Can you refer to https://github.com/testinprod-io/op-erigon/pull/99#issue-1976805494?

Reply:
oh I should read the above lines... got it. thank you 👍